### ISSUE TITLE
Disable debug names in production builds

### ISSUE THREAD

**Issue by mweststrate** [mobxjs/mobx#2737]
See https://github.com/mobxjs/mobx/discussions/2728

**Comment by iChenLei** [comment]
## Simple JS Heap Memory Test
#### wirte simple loop to create many mobx observable object
```js
import { configure, observable } from 'mobx';

configure({
  useProxies: "never"
});

const HugeRootMobx = {};
for (let i  = 0; i < 100000; i++) {  👈🏻   // Yeah, One hundred thousand loop
 HugeRootMobx[i] = observable({ val: i });
}

window.mobxTest = HugeRootMobx;
```
Add compiled js to html and open in chrome.

* Open chrome devtools
* Click Memory Tab
* Click `Take snapshot` button
* Save `HEAP SNAPSHOT`

#### Modify Mobx Source Code
Force `name_`(a.k.a `debug name`) to `void 0` everywhere.

```ts
// `Atom`、 `ComputedValue`、 `ObservableMap`、 `ObservableSet` and `ObservableValue`
name_ = void 0;
```
Run test again and save `HEAP SNAPSHOT`

#### Compare two JS Heap snapshot

![Snapshot](https://user-images.githubusercontent.com/14012511/105702163-54ccf600-5f46-11eb-99c6-9088430410b8.png)

I noticed that disable debugs name in production builds can save memory usage.(~16%)

## Simple test, not precise

If any inaccuracy, please mention it in this discussion line. 😃 

## More Think

It looks like implement this feature is meaningful. Maybe we need support mobx4 mobx5 and mobx6 both. 
But dose it break mobx-devtools user experience ? And any missing hidden problem ？

> Anyway, I think it's very meaningful for big data admin webapp which built on top of mobx. 
It's the same for 3d webapp 、Heavy js games and more.

 Look ashley-taylor's  public github activity (fork github.com/hazelcast/hazelcast),  I know why they need this feature. 😆 
```
ashley-taylor  6d ago

We have a rather large data structure.
Running the chrome heap dump we notice that the mobx debug names represent about 13% of our heap.
Would be nice if you could disable the names.
```

**Comment by FredyC** [comment]
Resolved in #2780
